### PR TITLE
Adding struct column tests

### PR DESCRIPTION
**Pull request by ishmandoo**
This PR adds some testing for combinations of nested struct and list columns.

### CODE REVIEW

**Comment by ishmandoo** [comment]
Didn't see any tests along these lines, but might have missed them.

File: py-polars/tests/test_df.py [hunk 1]
```diff
@@ -350,6 +350,33 @@ def test_init_records() -> None:
     assert df_xy.rows() == [(1, 2), (2, 1), (1, 2)]
 
 
+def test_struct_cols() -> None:
+    """Test that struct columns can be imported and work as expected."""
+
+    def build_struct_df(data):
```
Comment (cjermain) [inline py-polars/tests/test_df.py:356]: The build is failing on type checking. Do you need to add typing here? Try running `make pre-commit` in the `py-polars/` directory.

**Review by cjermain — COMMENTED** [review]
Thanks for adding these tests. Many of the struct tests are in [`test_struct.py`](https://github.com/pola-rs/polars/blob/master/py-polars/tests/test_struct.py).

**Comment by ishmandoo** [comment]
Moved the test to `test_struct.py`. It seems to test a super set of what was covered by `test_recursive_arrow_conversion`, so I removed that, but happy to revert it if desired.

**Comment by ritchie46** [comment]
Thanks @ishmandoo  and @cjermain

**Pull request merged**